packages/app/src/components-styled/age-demographic/age-demographic-coordinates.ts

```diff
@@ -21,44 +22,65 @@ export interface AgeDemographicCoordinates {
   ageGroupPercentageScale: ValueOf<ScaleTypeToD3Scale<any, any, any>>;
   infectedPercentageScale: ValueOf<ScaleTypeToD3Scale<any, any, any>>;
   ageGroupRangeScale: ScaleBand<string>;
-  ageGroupPercentagePoint: (value: NationalTestedPerAgeGroupValue) => any;
-  infectedPercentagePoint: (value: NationalTestedPerAgeGroupValue) => any;
-  ageGroupRangePoint: (value: NationalTestedPerAgeGroupValue) => any;
-  getTooltipCoordinates: GetTooltipCoordinates<NationalTestedPerAgeGroupValue>;
+  ageGroupPercentagePoint: (value: T) => any;
+  infectedPercentagePoint: (value: T) => any;
+  ageGroupRangePoint: (value: T) => any;
+  getTooltipCoordinates: GetTooltipCoordinates<T>;
   isSmallScreen: boolean;
   margin: {
     top: number;
     right: number;
     bottom: number;
     left: number;
   };
-  values: NationalTestedPerAgeGroupValue[];
-  ageGroupRange: (value: NationalTestedPerAgeGroupValue) => string;
+  values: T[];
+  ageGroupRange: (value: T) => string;
   ageRangeAxisWidth: number;
 }
 
-export function useAgeDemographicCoordinates(
-  data: NationalTestedPerAgeGroup,
-  isSmallScreen: boolean,
-  parentWidth: number,
-  isExtraSmallScreen: boolean
+export function useAgeDemographicCoordinates<
+  T extends AgeDemographicDefaultValue
+>(
+  data: { values: T[] },
+  metricProperty: keyof T,
+  visuallyMaxPercentage?: number
 ) {
-  return useMemo(() => {
+  const [ref, { width }] = useElementSize<HTMLDivElement>(400);
+  const { xs, xl } = useBreakpoints();
+  const isSmallScreen = !xl;
+  const isExtraSmallScreen = xs;
+
+  const coordinates = useMemo(() => {
     return calculateAgeDemographicCoordinates(
       data,
+      metricProperty,
       isSmallScreen,
-      parentWidth,
-      isExtraSmallScreen
+      width,
+      isExtraSmallScreen,
+      visuallyMaxPercentage
     );
-  }, [data, isSmallScreen, parentWidth, isExtraSmallScreen]);
+  }, [
+    data,
+    metricProperty,
+    isSmallScreen,
+    width,
+    isExtraSmallScreen,
+    visuallyMaxPercentage,
+  ]);
+
+  return [ref, coordinates] as const;
 }
 
-function calculateAgeDemographicCoordinates(
-  data: NationalTestedPerAgeGroup,
+function calculateAgeDemographicCoordinates<
+  T extends AgeDemographicDefaultValue
+>(
+  data: { values: T[] },
+  metricProperty: keyof T,
   isSmallScreen: boolean,
   parentWidth: number,
-  isExtraSmallScreen: boolean
-): AgeDemographicCoordinates {
+  isExtraSmallScreen: boolean,
```

Comment:
nit; 
```suggestion
  isTinyScreen: boolean,
```

Don't know, I just like the word :)

Reply:
And how do you feel about 'micro'? ;)

Reply:
I'll stick to extraSmall becasue it's based on the breakpoints.xs :-)

Reply:
Hmm less exiting to me,  but I see you are going for same character count 😄 OCD. I'm fine with that too, although I don't think you would normally call the screens that we are talking about "micro" in real life.